py-polars/polars/internals/frame.py

```diff
@@ -1925,7 +1961,8 @@ def __getitem__(
                 # df[:, [1, 2]]
                 # select by column indexes
                 if isinstance(col_selection[0], int):
-                    series_list = [self.to_series(i) for i in col_selection]
+                    # we only check the first element, but assume a homogenous list
+                    series_list = [self.to_series(cast(int, i)) for i in col_selection]
```

Comment:
You can avoid `cast` by changing the `if isinstance(  )` to ` if is_int_sequence(col_selection)`, which is a type guard in `utils.py`. That will slightly change the code, as it no longer checks just the first element of `col_selection` but all, but that is anyways the intention given how you typed it here.

Reply:
Nice. I was concerned about turning an `O(1)` check into `O(n)` but we loop through all the columns anyway...